src/spatial_reaction_systems/spatial_ODE_systems.jl

```diff
@@ -0,0 +1,211 @@
+### Spatial ODE Functor Structures ###
+
+# Functor structure containg the information for the forcing function of a spatial ODE with spatial movement on a lattice.
+struct LatticeDiffusionODEf{R,S,T}
+    ofunc::R
+    nV::Int64
+    nS::Int64
+    pV::Vector{Vector{Float64}}
+    work_pV::Vector{Float64}    
+    enumerated_pV_idx_types::Base.Iterators.Enumerate{BitVector}
+    spatial_rates::Vector{S}
+    leaving_rates::Matrix{Float64}
+    enumerated_edges::T
+    
+    function LatticeDiffusionODEf(ofunc::R, pV, spatial_rates::Vector{S}, lrs::LatticeReactionSystem) where {R, S, T}
+        leaving_rates = zeros(length(spatial_rates), lrs.nV)
+        for (s_idx, rates) in enumerate(last.(spatial_rates)),
+            (e_idx, e) in enumerate(edges(lrs.lattice))
+    
+            leaving_rates[s_idx, e.src] += get_component_value(rates, e_idx)
+        end
+        work_pV = zeros(lrs.nV)
+        enumerated_pV_idx_types = enumerate(length.(pV) .== 1)
+        enumerated_edges = deepcopy(enumerate(edges(lrs.lattice)))
+        new{R,S,typeof(enumerated_edges)}(ofunc, lrs.nV, lrs.nS, pV, work_pV, enumerated_pV_idx_types, spatial_rates, leaving_rates, enumerated_edges)
+    end
+end
+
+# Functor structure containg the information for the forcing function of a spatial ODE with spatial movement on a lattice.
+struct LatticeDiffusionODEjac{S,T}
+    ofunc::S
+    nV::Int64
+    nS::Int64
+    pV::Vector{Vector{Float64}}
+    work_pV::Vector{Float64}    
+    enumerated_pV_idx_types::Base.Iterators.Enumerate{BitVector}
```

Comment:
Sorry, I am not sure if I understand? 

I agree that the inability to change parameters mid-run is a weakness of the current approach.

Reply:
Or do you mean add it here in such a way that we can redo the Jacobian it uses. In case we want to have a callback that modified some diffusion rate parameter?